### ISSUE TITLE
Corrupted sf dataframes with osm_sf

### ISSUE THREAD

**Issue by polettif** [ropensci/osmdata#183]
When I load some data with `osm_sf` the osm_points dataframe is somehow corrupted. osm_ids are missing as a column and stored as rownames. The console output also doesn't really look like an sf dataframe.
``` r
library(osmdata)
#> Data (c) OpenStreetMap contributors, ODbL 1.0. http://www.openstreetmap.org/copyright
bbox = c(8.64851, 46.82073, 8.65284, 46.82368)
osm_qry = add_osm_feature(opq(bbox), key = "highway")
osm_sf = osmdata_sf(osm_qry)
colnames(osm_sf$osm_points)
#> [1] "highway"  "source"   "geometry"
head(osm_sf$osm_points)
#>           highway source            geometry
#> 24976723     <NA>   <NA> 8.647464, 46.829012
#> 24976725     <NA>   <NA> 8.647279, 46.828432
#> 24976726     <NA>   <NA> 8.647332, 46.827524
#> 24976728     <NA>   <NA>   8.64788, 46.82690
#> 24976730     <NA>   <NA> 8.649732, 46.825240
#> 359997770    <NA>   <NA> 8.649339, 46.825690
```

<sup>Created on 2019-08-22 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Something similar happens with osm_lines:
``` r
library(osmdata)
#> Data (c) OpenStreetMap contributors, ODbL 1.0. http://www.openstreetmap.org/copyright
bbox = c(8.64851, 46.82073, 8.65284, 46.82368)
osm_qry = add_osm_feature(opq(bbox), key = "highway")
osm_sf = osmdata_sf(osm_qry)
head(osm_sf$osm_lines)
#>              osm_id               name embankment foot     highway  lit
#> 23725641   23725641    Gotthardstrasse       <NA> <NA>     primary <NA>
#> 99184746   99184746      Bärenbodenweg       <NA> <NA> residential <NA>
#> 99202608   99202608 Schlossbergstrasse         no <NA>    tertiary   no
#> 100809880 100809880 Schlossbergstrasse         no <NA>    tertiary   no
#> 106632651 106632651   Friedheimstrasse       <NA> <NA> residential <NA>
#> 106632653 106632653          Butzenweg       <NA> <NA> residential <NA>
#>           maxheight maxspeed oneway  ref source surface tunnel
#> 23725641       <NA>       50   <NA>    2   <NA>    <NA>   <NA>
#> 99184746       <NA>     <NA>   <NA> <NA>   <NA> asphalt   <NA>
#> 99202608        2.7       30   <NA> <NA>   <NA> asphalt    yes
#> 100809880      <NA>     <NA>   <NA> <NA>   <NA> asphalt   <NA>
#> 106632651      <NA>     <NA>   <NA> <NA>   <NA> asphalt   <NA>
#> 106632653      <NA>     <NA>   <NA> <NA>   <NA> asphalt   <NA>
#>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         geometry
#> 23725641  8.647464, 8.647429, 8.647298, 8.647279, 8.647232, 8.647196, 8.647266, 8.647332, 8.647485, 8.647614, 8.647880, 8.648132, 8.648181, 8.648827, 8.649339, 8.649500, 8.649533, 8.649624, 8.649732, 8.649798, 8.649839, 8.649970, 8.650012, 8.650045, 8.650102, 8.650121, 8.650229, 8.650281, 8.650298, 8.650312, 8.650353, 8.650355, 8.650368, 8.650358, 8.650367, 8.650361, 8.650364, 8.650394, 8.650402, 8.650420, 46.829012, 46.828902, 46.828494, 46.828432, 46.828177, 46.827934, 46.827654, 46.827524, 46.827286, 46.827151, 46.826898, 46.826685, 46.826645, 46.826125, 46.825690, 46.825529, 46.825488, 46.825377, 46.825240, 46.825082, 46.824987, 46.824602, 46.824478, 46.824386, 46.824182, 46.824123, 46.823779, 46.823613, 46.823536, 46.823459, 46.823022, 46.822676, 46.822310, 46.822232, 46.822054, 46.821837, 46.821655, 46.821208, 46.821117, 46.820938
#> 99184746                                                                                                                                                                                                                    8.654250, 8.654372, 8.654674, 8.654766, 8.654772, 8.654764, 8.654697, 8.654602, 8.654390, 8.654277, 8.654131, 8.654035, 8.653997, 8.653939, 8.653720, 8.653717, 8.653709, 8.653693, 8.653594, 8.653420, 8.653235, 8.653138, 8.653092, 8.652987, 8.652894, 8.652748, 8.652662, 8.652345, 8.652252, 8.652226, 46.817118, 46.817226, 46.817565, 46.817854, 46.818123, 46.818357, 46.818487, 46.818615, 46.818739, 46.818767, 46.818837, 46.818978, 46.818993, 46.819003, 46.818968, 46.819135, 46.819233, 46.819371, 46.819525, 46.819656, 46.819806, 46.820001, 46.820071, 46.820233, 46.820451, 46.820634, 46.820740, 46.820930, 46.820976, 46.821040
#> 99202608                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                8.649734, 8.650337, 46.820688, 46.820877
#> 100809880                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               8.650337, 8.650420, 46.820877, 46.820938
#> 106632651                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     8.652001, 8.651229, 8.650982, 8.650361, 46.822153, 46.821885, 46.821834, 46.821837
#> 106632653                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       8.651991, 8.651924, 8.651906, 8.651934, 8.652003, 8.652088, 8.652120, 8.652288, 8.652278, 8.652001, 46.820959, 46.821043, 46.821109, 46.821166, 46.821220, 46.821321, 46.821373, 46.821689, 46.821781, 46.822153
```

<sup>Created on 2019-08-22 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by mpadge** [comment]
Thanks @poletif, esp for the nice reprex. A similar thing arose recently in another case too, and was due to some ill-formed data at the OSM level, which the `osmdata` code failed to process adequately. I'll check this case out and fix asap

**Comment by mpadge** [comment]
Oh, actually, it has been fixed, so if you just `removes::install_github()` to latest `osmdata`, then all should be good. As for console output, the `print` method for `sf` only works with `sf` loaded into namespace, so if you've only got `osmdata` loaded but not `sf`, then output won't look like expected `sf` output. Please confirm that update fixes, and we'll close. (And this case indicates the general usefulness of including a `packageVersion("osmdata")` in your reprex, the problem should be fixed in `0.1.1.4`, and i guess yours is less than that?) Thanks for the input!

**Comment by polettif** [comment]
Thanks, using the newest version from github did fix the problem. I was using the CRAN version 0.1.1.

> As for console output, the print method for sf only works with sf loaded into namespace, so if you've only got osmdata loaded but not sf, then output won't look like expected sf output.

Ah I see, good to know!
